<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>AutoTaskPage</class>
 <widget class="QWidget" name="AutoTaskPage">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>800</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>自动任务</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QHBoxLayout" name="topBarLayout">
     <item>
      <widget class="QLabel" name="lbl_task_status">
       <property name="text">
        <string>状态: 空闲</string>
       </property>
       <property name="font">
        <font>
         <pointsize>10</pointsize>
         <weight>75</weight>
         <bold>true</bold>
        </font>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btn_load_task">
       <property name="text">
        <string>加载任务</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btn_reload">
       <property name="text">
        <string>重新加载</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="mainContentLayout">
     <item>
      <widget class="QGroupBox" name="group_task_list">
       <property name="title">
        <string>任务列表</string>
       </property>
       <property name="minimumSize">
        <size>
         <width>250</width>
         <height>0</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>300</width>
         <height>16777215</height>
        </size>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QListWidget" name="list_tasks"/>
        </item>
        <item>
         <widget class="QLabel" name="lbl_task_info">
          <property name="text">
           <string>未选择任务</string>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btn_import_task">
          <property name="text">
           <string>导入任务...</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="group_execution">
       <property name="title">
        <string>执行状态</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QLabel" name="lbl_current_step">
          <property name="text">
           <string>当前步骤: --</string>
          </property>
          <property name="font">
           <font>
            <pointsize>10</pointsize>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QProgressBar" name="progress_step">
          <property name="value">
           <number>0</number>
          </property>
          <property name="textVisible">
           <bool>true</bool>
          </property>
          <property name="format">
           <string>步骤进度: %p%</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QProgressBar" name="progress_total">
          <property name="value">
           <number>0</number>
          </property>
          <property name="format">
           <string>总体进度: %p%</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTableWidget" name="table_steps">
          <property name="editTriggers">
           <set>QAbstractItemView::NoEditTriggers</set>
          </property>
          <property name="selectionMode">
           <enum>QAbstractItemView::SingleSelection</enum>
          </property>
          <property name="selectionBehavior">
           <enum>QAbstractItemView::SelectRows</enum>
          </property>
          <column>
           <property name="text">
            <string>#</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>类型</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>目标</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>预设</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>状态</string>
           </property>
          </column>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="group_realtime">
          <property name="title">
           <string>实时信息</string>
          </property>
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>100</height>
           </size>
          </property>
          <layout class="QGridLayout" name="gridLayout">
           <item row="0" column="0">
            <widget class="QLabel" name="label_depth">
             <property name="text">
              <string>当前深度:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLabel" name="lbl_depth">
             <property name="text">
              <string>0.0 mm</string>
             </property>
             <property name="font">
              <font>
               <weight>75</weight>
               <bold>true</bold>
              </font>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_target">
             <property name="text">
              <string>目标深度:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLabel" name="lbl_target">
             <property name="text">
              <string>0.0 mm</string>
             </property>
             <property name="font">
              <font>
               <weight>75</weight>
               <bold>true</bold>
              </font>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_elapsed">
             <property name="text">
              <string>已用时间:</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QLabel" name="lbl_elapsed">
             <property name="text">
              <string>00:00</string>
             </property>
             <property name="font">
              <font>
               <weight>75</weight>
               <bold>true</bold>
              </font>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="group_presets">
       <property name="title">
        <string>参数预设</string>
       </property>
       <property name="minimumSize">
        <size>
         <width>300</width>
         <height>0</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>400</width>
         <height>16777215</height>
        </size>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <widget class="QTableWidget" name="table_presets">
          <property name="editTriggers">
           <set>QAbstractItemView::NoEditTriggers</set>
          </property>
          <column>
           <property name="text">
            <string>ID</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Vp</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>RPM</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Fi</string>
           </property>
          </column>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="lbl_current_preset">
          <property name="text">
           <string>当前预设: --</string>
          </property>
          <property name="font">
           <font>
            <pointsize>9</pointsize>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="controlBarLayout">
     <item>
      <widget class="QPushButton" name="btn_start">
       <property name="text">
        <string>开始</string>
       </property>
       <property name="styleSheet">
        <string notr="true">QPushButton { background-color: #67c23a; color: white; font-weight: bold; min-height: 40px; }</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btn_pause">
       <property name="text">
        <string>暂停</string>
       </property>
       <property name="styleSheet">
        <string notr="true">QPushButton { background-color: #e6a23c; color: white; font-weight: bold; min-height: 40px; }</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btn_resume">
       <property name="text">
        <string>恢复</string>
       </property>
       <property name="styleSheet">
        <string notr="true">QPushButton { background-color: #409eff; color: white; font-weight: bold; min-height: 40px; }</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btn_stop">
       <property name="text">
        <string>停止</string>
       </property>
       <property name="styleSheet">
        <string notr="true">QPushButton { background-color: #909399; color: white; font-weight: bold; min-height: 40px; }</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_2">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="btn_emergency">
       <property name="text">
        <string>急停</string>
       </property>
       <property name="styleSheet">
        <string notr="true">QPushButton { background-color: #f56c6c; color: white; font-weight: bold; font-size: 14pt; min-width: 150px; min-height: 50px; border-radius: 8px; }</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QGroupBox" name="group_execution_records">
     <property name="title">
      <string>执行记录</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_records">
      <item>
       <layout class="QHBoxLayout" name="recordFilterLayout">
        <item>
         <widget class="QLabel" name="label_round_filter">
          <property name="text">
           <string>轮次</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="combo_round_filter"/>
        </item>
        <item>
         <widget class="QLabel" name="label_task_filter">
          <property name="text">
           <string>任务文件</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="combo_task_filter"/>
        </item>
        <item>
         <spacer name="horizontalSpacer_records">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="btn_refresh_records">
          <property name="text">
           <string>刷新记录</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QTableWidget" name="table_execution_records">
        <column>
         <property name="text">
          <string>时间</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>状态</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>步骤</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>深度(mm)</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>扭矩(Nm)</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>钻压(N)</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>原因</string>
         </property>
        </column>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" name="recordSummaryLayout">
        <item>
         <widget class="QLabel" name="lbl_record_summary">
          <property name="text">
           <string>未选择记录</string>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="lbl_sensor_snapshot">
          <property name="text">
           <string>传感器摘要将显示在此处</string>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="group_log">
     <property name="title">
      <string>日志</string>
     </property>
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>150</height>
      </size>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_5">
      <item>
       <widget class="QTextBrowser" name="tb_log">
        <property name="styleSheet">
         <string notr="true">QTextBrowser { background-color: #2c3e50; color: #ecf0f1; font-family: Consolas, monospace; }</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
